Case study for ZCITC Streamlining Urban Parking with KubeEdge (ENG) (#659)
kubeedge/website: src/pages/case-studies/zcitc/index.mdx

adding complexity to the technical stack and increasing management challenges for development teams.

Zhejiang Zhicheng Software Co., Ltd. specializes in city-wide smart parking solutions, leveraging big data for intelligent parking resource integration. 
To address the needs of unattended parking lots for high security, reliability, efficiency, and maintainability, the Zhicheng team increasingly deploys 
containerized applications on edge nodes at parking lots.

# Cloud-Native Edge Computing Technology

Edge computing is a distributed architecture that processes data closer to the user, reducing latency and improving transmission speeds compared to cloud computing. 
Cloud-native methods build and run applications using open-source technologies like Kubernetes and Docker. They rely on microservice architectures, 
agile methodologies, and DevOps practices to enable flexibility, scalability, and automation.

As computing demands grow, combining cloud-native and edge computing creates a new technical trend: cloud-native edge computing. 
This approach brings cloud capabilities to edge devices, enabling offline operations, edge-cloud collaboration, and massive device connectivity. 

> KubeEdge is a leading solution implementing this paradigm, extending Kubernetes’ capabilities to the edge.

Reviewer: There seems to be something wrong with the format here. It seems like there are a few extra spaces.
![image](https://github.com/user-attachments/assets/c34e1d0e-095a-4976-845e-1f3823e4705f)

Author: ![image](https://github.com/user-attachments/assets/c22f8e06-5119-49e4-a0db-b042406f4f14)

this appears fine in the preview, and I confirmed no extra spaces are there.

   For large-scale parking lots, dual or multiple machine hot-standby solutions are implemented. PostgreSQL and Redis applications use StatefulSet mechanisms for stable re-scheduling.

![Master-Slave Configuration](images/MasterSlave.png)

2. **Multi-Active Redundancy:**
   Kubernetes ReplicaSet ensures multiple application replicas are active simultaneously for load balancing and high availability.

3. **Node Deployment of Related Applications:**
   Parking management applications and device control services are co-deployed to optimize communication and response times.

4. **Cross-Node Deployment of Application Instances:**
   Instances of the same application are deployed across nodes to improve availability.

5. **Grouped Deployment Based on Edge Node Properties:**
   Applications are deployed based on edge node labels for different CPU architectures or device types.

6. **Automatic Deployment of New Nodes:**
   Newly added nodes are labeled, and applications are automatically deployed.

7. **Inter-Linked Parking Lots:**
   Complex scenarios requiring communication between parking lots use KubeEdge’s EdgeMesh for lightweight, integrated service discovery.

## Results

> Currently, Zhicheng's smart parking operating system has undergone domestic innovation transformation, with containerized applications fully adapted to computing platforms with x86_64, 
> aarch32, and aarch64 CPU architectures, elastically adapting to parking lot scale and construction costs. Gateway devices use lightweight MQTT protocol for message communication, 

Reviewer: ```suggestion
Currently, Zhicheng's smart parking operating system has implemented application containerization, fully adapted to the computing platform of x86_64, aarch32, and aarch64 CPU architecture, elastically adapting to parking lot scale and construction costs. Gateway devices use lightweight MQTT protocol for message communication, 
```

achieving low consumption and stability while greatly reducing bandwidth pressure. Kubernetes' container orchestration capabilities bring rapid deployment and high availability to 
parking lot systems.

In practice, implementation teams can deploy one or several edge gateway devices with appropriate computing power as edge computing platforms according to parking lot size and turnover rate, 
following a "one block, one strategy, one solution" approach. Edge computing provides the following capabilities for the cloud platform:

> (1) Intelligent access to parking lot devices. Achieves collection and reporting of sensor data from entrance/exit cameras and vehicle detectors, and networked control of barriers, 
> space display screens, voice broadcasting, and other control units.
> (2) Two-way synchronization of business data. Real-time communication with the cloud, receiving basic information like parking lot, channel, and vehicle whitelist information from the cloud; 
> pre-processing vehicle entry/exit information, generating parking orders, and calculating fees based on charging schemes. Business data has two-way synchronization capability while ensuring data consistency.

Reviewer: ```suggestion
(1) Intelligent access to parking lot devices. Achieves collection and reporting of sensor data from entrance/exit cameras and vehicle detectors, and networked control of barriers, 
space display screens, voice broadcasting, and other control units.

(2) Two-way synchronization of business data. Real-time communication with the cloud, receiving basic information like parking lot, channel, and vehicle whitelist information from the cloud; 
pre-processing vehicle entry/exit information, generating parking orders, and calculating fees based on charging schemes. Business data has two-way synchronization capability while ensuring data consistency.
```
You can add a blank line between two paragraphs to make the format of the article clearer.

![image](https://github.com/user-attachments/assets/b005754a-afed-424c-9911-367474f37c12)

Author: Thanks @wbc6080 for reviewing, added your suggestions (actually this I used because the zh version also had same formatting.)
